### PR TITLE
Cargo cleanup

### PR DESCRIPTION
**Pull request by rkuris**
Removed some unused dependencies from the main project.

### CODE REVIEW

**Review by iFrostizz — CHANGES_REQUESTED** [review]
There are some unused dependencies left. You can run [`cargo machete`](https://github.com/bnjbvr/cargo-machete) to list them automatically, though there is a false positive for `serde` and `prost`.

```diff
diff --git a/grpc-testtool/Cargo.toml b/grpc-testtool/Cargo.toml
index f7a22fe..24843f6 100644
--- a/grpc-testtool/Cargo.toml
+++ b/grpc-testtool/Cargo.toml
@@ -18,12 +18,10 @@ bench = false
 [dependencies]
 firewood = { version = "0.0.4", path = "../firewood" }
 prost = "0.12.3"
-thiserror = "1.0.57"
 tokio = { version = "1.36.0", features = ["sync", "rt-multi-thread"] }
 tonic = { version = "0.11.0", features = ["tls"] }
 tracing = { version = "0.1.40" }
 clap = { version = "4.5.0", features = ["derive"] }
-tempdir = "0.3.7"
 log = "0.4.20"
 env_logger = "0.11.2"
 chrono = "0.4.34"
diff --git a/storage/Cargo.toml b/storage/Cargo.toml
index b4efac6..da7356e 100644
--- a/storage/Cargo.toml
+++ b/storage/Cargo.toml
@@ -13,8 +13,6 @@ hex = "0.4.3"
 serde = { version = "1.0.199", features = ["derive"] }
 smallvec = { version = "1.13.2", features = ["serde", "write", "union"] }
 sha2 = "0.10.8"
-bytemuck = { version = "1.14.3", features = ["derive"] }
-integer-encoding = "4.0.0"

 [dev-dependencies]
 rand = "0.8.5"
```

**Comment by rkuris** [comment]
> use `cargo machete`

Great suggestion, thanks! I left serde alone because I think we might be able to remove it completely later.

File: storage/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ hex = "0.4.3"
 serde = { version = "1.0.199", features = ["derive"] }
 smallvec = { version = "1.13.2", features = ["serde", "write", "union"] }
 sha2 = "0.10.8"
-bytemuck = { version = "1.14.3", features = ["derive"] }
+integer-encoding = "4.0.0"
```
Comment (iFrostizz) [inline storage/Cargo.toml:16]: Looks like you left this guy

**Review by iFrostizz — APPROVED** [review]

File: storage/Cargo.toml (hunk 1 above)
Comment (rkuris) [reply storage/Cargo.toml:16]: Yeah, we'll need it very soon as the one user (hashednode.rs) will be moving to storage very soon.

**Pull request merged**